In [22]:
!pip install yfinance
# add all installable libs

## Fetching data

In [23]:
import yfinance as yf
import pandas as pd

# Define ticker and date range
ticker = 'ETH-USD'
start_date = '2017-01-01'
end_date = '2024-12-31'

# Download data
eth_data = yf.download(ticker, start=start_date, end=end_date, interval='1d')

# Reset index to make 'Date' a column
eth_data.reset_index(inplace=True)

# Display the first few rows
print(eth_data.head())

# Save to CSV for future use
eth_data.to_csv('eth_usd_2017_2024.csv', index=False)

[*********************100%***********************]  1 of 1 completed

Price        Date       Close        High         Low        Open      Volume
Ticker                ETH-USD     ETH-USD     ETH-USD     ETH-USD     ETH-USD
0      2017-11-09  320.884003  329.451996  307.056000  308.644989   893249984
1      2017-11-10  299.252991  324.717987  294.541992  320.670990   885985984
2      2017-11-11  314.681000  319.453003  298.191986  298.585999   842300992
3      2017-11-12  307.907990  319.153015  298.513000  314.690002  1613479936
4      2017-11-13  316.716003  328.415009  307.024994  307.024994  1041889984


In [24]:
eth_data.shape

(2609, 6)

In [25]:
eth_data.info() 
# no null vals + no cat cols

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2609 entries, 0 to 2608
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   (Date, )           2609 non-null   datetime64[ns]
 1   (Close, ETH-USD)   2609 non-null   float64       
 2   (High, ETH-USD)    2609 non-null   float64       
 3   (Low, ETH-USD)     2609 non-null   float64       
 4   (Open, ETH-USD)    2609 non-null   float64       
 5   (Volume, ETH-USD)  2609 non-null   int64         
dtypes: datetime64[ns](1), float64(4), int64(1)
memory usage: 122.4 KB


In [26]:
eth_data.head()

Price        Date       Close        High         Low        Open      Volume
Ticker                ETH-USD     ETH-USD     ETH-USD     ETH-USD     ETH-USD
0      2017-11-09  320.884003  329.451996  307.056000  308.644989   893249984
1      2017-11-10  299.252991  324.717987  294.541992  320.670990   885985984
2      2017-11-11  314.681000  319.453003  298.191986  298.585999   842300992
3      2017-11-12  307.907990  319.153015  298.513000  314.690002  1613479936
4      2017-11-13  316.716003  328.415009  307.024994  307.024994  1041889984

In [27]:
eth_data.tail()

Price        Date        Close         High          Low         Open  \
Ticker                 ETH-USD      ETH-USD      ETH-USD      ETH-USD   
2604   2024-12-26  3331.225830  3512.604492  3302.306396  3493.304199   
2605   2024-12-27  3328.916992  3436.710693  3302.575684  3331.053711   
2606   2024-12-28  3397.902344  3419.920166  3318.033936  3328.774658   
2607   2024-12-29  3349.513428  3406.648438  3321.664795  3397.862549   
2608   2024-12-30  3356.392578  3428.527344  3298.804443  3349.585938   

Price        Volume  
Ticker      ETH-USD  
2604    22247726776  
2605    24091627403  
2606    14305648523  
2607    13440907792  
2608    26981583962

In [28]:
eth_data.describe() 

Price                  Date        Close         High          Low  \
Ticker                           ETH-USD      ETH-USD      ETH-USD   
count                  2609  2609.000000  2609.000000  2609.000000   
mean    2021-06-05 00:00:00  1490.773764  1530.573913  1445.274314   
min     2017-11-09 00:00:00    84.308296    85.342743    82.829887   
25%     2019-08-23 00:00:00   268.704041   274.320831   261.960510   
50%     2021-06-05 00:00:00  1352.837158  1390.589966  1308.991455   
75%     2023-03-19 00:00:00  2405.181152  2472.187744  2322.337158   
max     2024-12-30 00:00:00  4812.087402  4891.704590  4718.039062   
std                     NaN  1226.266551  1258.805914  1189.917204   

Price          Open        Volume  
Ticker      ETH-USD       ETH-USD  
count   2609.000000  2.609000e+03  
mean    1489.776234  1.301809e+10  
min       84.279694  6.217330e+08  
25%      268.692261  5.556444e+09  
50%     1353.205078  1.061841e+10  
75%     2401.256348  1.768550e+10  
max     4810.071289  8.448291e+10  
std     1226.214741  1.059053e+10


* **Time Covered:** This data shows us the price of Ethereum and how much it was traded each day from late 2017 to the end of 2024.
* **How Much Data:** 2600 individual days.
* **Average End-of-Day Price:** If you look at the price of Ethereum at the end of each day, the average over this whole period was about $1490.
* **Price Swing:** The price went as low as about $84 on some days and as high as over $4800 on other days. This shows how much the price of Ethereum could go up or down.
* The **average price** (mean) over all the days is around $1490.
* The **middle price** (median) of all the daily prices is around $1352. This means that for roughly half of the days, the price was below $1352, and for roughly half, it was above.
* **Price Changeability:** The price of Ethereum changed quite a lot day-to-day. The standard deviation of around $1226 tells us that the price often moved significantly away from the average.

In [ ]:
eth_data.columns = [col[0] if isinstance(col, tuple) else col for col in eth_data.columns]
eth_data.head()

Date       Close        High         Low        Open      Volume
0 2017-11-09  320.884003  329.451996  307.056000  308.644989   893249984
1 2017-11-10  299.252991  324.717987  294.541992  320.670990   885985984
2 2017-11-11  314.681000  319.453003  298.191986  298.585999   842300992
3 2017-11-12  307.907990  319.153015  298.513000  314.690002  1613479936
4 2017-11-13  316.716003  328.415009  307.024994  307.024994  1041889984

In [ ]:
import plotly.express as px

px.line(eth_data, x=eth_data.index, y='Close', title='Ethereum (ETH/USDT) Daily Closing Price')

In [31]:
px.line(eth_data, x=eth_data.index, y='Volume', title='Ethereum (ETH/USDT) Daily Trading Volume')

In [32]:
import plotly.express as px

eth_data['rolling_mean_30'] = eth_data['Close'].rolling(window=30).mean()
eth_data['rolling_std_30'] = eth_data['Close'].rolling(window=30).std()

fig = px.line(eth_data,
              x=eth_data.index,
              y=['Close', 'rolling_mean_30'],
              title='Daily Price with Monthly Average Price',
              labels={'Close': 'Daily Price', 'rolling_mean_30': 'Monthly Average Price'})
fig.show()

In [33]:
px.line(eth_data, x=eth_data.index, y='rolling_std_30', title='30-Day Rolling Standard Deviation (Volatility)')

In [34]:
px.box(eth_data, y='Close', title='Ethereum Closing Price Distribution')